File: app/src/main/java/com/amary/app/data/moviecat/activity/MainActivity.java
package com.amary.app.data.moviecat.activity;

import android.content.res.Configuration;
import android.os.Bundle;

import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import com.amary.app.data.moviecat.adapter.MenuTabAdapter;
import com.amary.app.data.moviecat.base.BaseActivity;
import com.google.android.material.tabs.TabLayout;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.amary.app.data.moviecat.R.id;
import static com.amary.app.data.moviecat.R.layout;

public class MainActivity extends BaseActivity {

    @BindView(id.tab_menu)
    TabLayout tabMenu;
    @BindView(id.fl_container)
    ViewPager viewMenu;
    @BindView(id.toolbar_main)
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(layout.activity_main);

        ButterKnife.bind(this);
        setSupportActionBar(toolbar);

        MenuTabAdapter adapter = new MenuTabAdapter(getSupportFragmentManager(), tabMenu.getTabCount());
        viewMenu.setAdapter(adapter);
        viewMenu.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabMenu));
        tabMenu.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewMenu.setCurrentItem(tab.getPosition());
            }
            @Override
            public void onTabUnselected(TabLayout.Tab tab) { }
            @Override
            public void onTabReselected(TabLayout.Tab tab) { }
        });


    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }

}

File: app/src/main/java/com/amary/app/data/moviecat/activity/MainActivity.kt
package com.amary.app.data.moviecat.activity

import android.os.Bundle
import com.google.android.material.bottomnavigation.BottomNavigationView
import androidx.fragment.app.Fragment
import androidx.appcompat.app.AppCompatActivity
import com.amary.app.data.moviecat.R
import com.amary.app.data.moviecat.activity.fragment.MoviesFragment
import com.amary.app.data.moviecat.activity.fragment.TvShowFragment
import java.util.*

class MainActivity : AppCompatActivity() {

    private val onNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { menuItem ->
        lateinit var fragment: Fragment
        when (menuItem.itemId) {
            R.id.nv_movie -> {
                //fragment movie
                fragment = MoviesFragment()
                supportFragmentManager
                        .beginTransaction()
                        .replace(R.id.fl_container, fragment)
                        .commit()
                supportActionBar?.setTitle(R.string.tvshow)

                return@OnNavigationItemSelectedListener true
            }
            R.id.nv_tvshow -> {
                //fragment tvshow
                fragment = TvShowFragment()
                supportFragmentManager
                        .beginTransaction()
                        .replace(R.id.fl_container, fragment)
                        .commit()
                supportActionBar?.setTitle(R.string.tvshow)

                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val naviBottom = findViewById<BottomNavigationView>(R.id.bnv_layout)
        naviBottom.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener)

        if (savedInstanceState == null) {
            naviBottom.selectedItemId = R.id.nv_movie
            Objects.requireNonNull(supportActionBar)!!.setTitle(R.string.movies)
        }
    }
}